mathics/builtin/datentime.py

```diff
@@ -157,6 +158,48 @@ def evaluate(self, evaluation):
 
 
 class _DateFormat(Builtin):
+    messages = {
+        'arg': 'Argument `1` cannot be interpreted as a date or time input.',
+        'str': 'String `1` cannot be interpreted as a date in format `2`.',
+        'ambig': 'The interpretation of `1` is ambiguous.',
+        'fmt': '`1` is not a valid date format.',
+    }
+
+    automatic = re.compile(r'^([0-9]{1,4})([^0-9]+)([0-9]{1,2})\2([0-9]{1,4})\s*')
```

Comment:
I'm wondering if repeating the separator is ideal, I'm thinking of e.g. `2016  09 17` (two spaces then one).